app/charts/bar/bars-stacked-state.tsx

```diff
@@ -174,53 +174,53 @@ const useBarsStackedState = (
     getSegment,
   ]);
 
-  const sumsByX = useMemo(() => {
+  const sumsByY = useMemo(() => {
     return Object.fromEntries(
       rollup(
         chartData,
-        (v) => sum(v, (x) => getY(x)),
-        (x) => getX(x)
+        (v) => sum(v, (x) => getX(x)),
+        (x) => getY(x)
       )
     );
   }, [chartData, getX, getY]);
 
   const normalize = calculationType === "percent";
-  const chartDataGroupedByX = useMemo(() => {
+  const chartDataGroupedByY = useMemo(() => {
     if (normalize) {
       return group(
-        normalizeData(chartData, {
-          yKey: yMeasure.id,
-          getY,
-          getTotalGroupValue: (d) => sumsByX[getX(d)],
+        normalizeDataInverted(chartData, {
+          xKey: xMeasure.id,
+          getX,
+          getTotalGroupValue: (d) => sumsByY[getY(d)],
         }),
```

Comment:
Nit: I think here we should be safe to make the `normalizeData` function more generic and accept `valueKey` (instead of `xKey`), `getValue` (instead of `getX`) so that we can use it everywhere, in columns and bars, instead of having a new function 👀